lib/livebook/update_check.ex

```diff
@@ -0,0 +1,146 @@
+defmodule Livebook.UpdateCheck do
+  @moduledoc false
+
+  # Periodically checks for available Livebook update.
+
+  use GenServer
+
+  require Logger
+
+  @name __MODULE__
+  @timeout :infinity
+
+  @hour_in_ms 60 * 60 * 1000
+  @day_in_ms 24 * @hour_in_ms
+
+  @doc false
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, {}, name: @name)
+  end
+
+  @doc """
+  Returns the latest Livebook version if it's more recent than the
+  current one.
+  """
+  @spec new_version() :: String.t() | nil
+  def new_version() do
+    GenServer.call(@name, :get_new_version, @timeout)
+  end
+
+  @doc """
+  Returns whether the update check is enabled.
+  """
+  @spec enabled?() :: boolean()
+  def enabled?() do
+    GenServer.call(@name, :get_enabled, @timeout)
+  end
+
+  @doc """
+  Sets whether the update check is enabled.
+  """
+  @spec set_enabled(boolean()) :: :ok
+  def set_enabled(enabled) do
+    GenServer.cast(@name, {:set_enabled, enabled})
+  end
+
+  @impl true
+  def init({}) do
+    send(self(), :check)
+
+    {:ok,
+     %{
+       enabled: Livebook.Settings.update_check_enabled?(),
+       new_version: nil,
+       timer_ref: nil
+     }}
+  end
+
+  @impl true
+  def handle_cast({:set_enabled, enabled}, state) do
+    Livebook.Settings.set_update_check_enabled(enabled)
+
+    state = cancel_timer(state)
+
+    if enabled do
+      send(self(), :check)
```

Comment:
I would do `Process.send_after(self(), :check, 0)` so we always go through the same API. You can have an API called `schedule_timer` that receives the state and the time as arguments.

Reply:
Sounds good, done!